sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager.swift

```diff
@@ -430,7 +430,8 @@ internal final class URLSessionTransferManager: NSObject, TransferManager, URLSe
         case let transfer as BlockTransfer:
             operationQueue.add(BlockOperation(withTransfer: transfer, delegate: self))
         case let transfer as BlobTransfer:
-            transfer.progressHandler = progressHandler
+            // if progress handler not provided, do not overwrite with nil
+            transfer.progressHandler = progressHandler ?? transfer.progressHandler
```

Comment:
Prevents the reachability listener from wiping out the attached progress handler when calling `resumeAll`.